File: client/src/components/validation.js


import { maskSensitiveData } from './utils';
export const validateForm =(formData) =>{
  console.log("validateForm received:", maskSensitiveData (formData));

  if(!formData){
    return{ valid:false, message:"Missing data"}
  }
  const { firstname, surname, email, password} = formData;

  if (!email?.trim() || !password?.trim()){ 
    return {valid: false, message:"Email and password required"};
 
 }
 //signup validation 
  if (
    ("firstname" in formData && !firstname?.trim()) || 
    ("surname" in formData && !surname?.trim())
 ){
  return{valid: false, message:"First name and surname are required."};
  }

//email regex
  const emailRegex = /^[^@\s]+@[^@\s]+\.[^@\s]+$/;
  if (!emailRegex.test(formData.email.trim())){
   return{valid: false, message:"Please enter a valid email format."};
    
  }

//password
  const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{8,}$/;
  if (!passwordRegex.test(password.trim())){
     return{valid: false, 
      message:
      'Pasword must be atleast 8 characters, include uppercase, lowercase and a number.'};
          
  }
//Pass
  
  return {valid:true, message:""};

};

File: server/validation.py
import re 
#validation.py

def validate_signup_data(data):
    if not isinstance(data, dict):
        return False, "invalid input format. Expected JSON objects"
    
    required_fields =('firstname', 'surname', 'email' ,'password')
    if not all(k in data for k in required_fields):
        return False, "missing required fields"

    firstname = data.get("firstname", "").strip()
    surname = data.get("surname","").strip()
    email = data.get("email","").strip().lower()
    password =data.get("password","").strip()

    if not email or not password:
        return False, "Email and password required"

    if not firstname or not surname:
        return False,"firstname and surname required"
    
    email_regex= r'^[^@\s]+@[^@\s]+\.[^@\s]+$'
    if not re.match(email_regex, email):
        return False, "Please enter a valid email address"

    password_regex = r'^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{8,}$'
    if not re.match(password_regex, password):
        return False,("Password must be atleast 8 characters , include " "an uppercase, lowercase , and a number")
  
    return True, "valid"